app/src/main/java/com/example/chrono/main/circuit/CircuitTimerActivity.kt

```diff
@@ -0,0 +1,163 @@
+            // Make sure that the timer is shut down
+            countdown.cancel()
+            setResult(Activity.RESULT_CANCELED)
+            finish()
+        }
+    }
+
+    private fun createTimer(seconds: Int, wasPaused: Boolean) {
+        var time = seconds.toFloat() * 1000 + 250
+        if (wasPaused) {
+            time = secondsLeft * 1000 + 250
+        }
+        countdown = object : CountDownTimer(time.toLong(), 250) {
+            override fun onTick(p0: Long) {
+                if ((p0.toFloat().roundToInt() / 1000.0f) != secondsLeft) {
+                    secondsLeft = (p0.toFloat() / 1000.0f).roundToInt().toFloat()
+                    updateTimerUI()
+                }
+            }
+
+            override fun onFinish() {
+                if (currentSet != 0) {
+                    when (runningState) {
+                        RunningState.INIT -> {
+                            workout()
+                        }
+                        RunningState.WORK -> {
+                            rest()
+                        }
+                        RunningState.REST -> {
+                            workout()
+                        }
+                    }
+                } else {
+                    isDone()
+                }
+            }
+        }.start()
+    }
+
+    private fun isDone() {
+        Toast.makeText(this, "It's done", Toast.LENGTH_SHORT).show()
+        timerState = TimerState.INIT
+        runningState = RunningState.INIT
+        updateButtonUI()
+    }
+
+    // Update UI for every tick, possibly need to do more in the future
+    fun updateTimerUI() {
+        bind!!.countdown.text = (secondsLeft).toInt().toString()
+    }
+
+    private fun updateButtonUI() {
+        when (timerState) {
+            TimerState.INIT -> {
+                bind!!.initButtonLay.visibility = View.VISIBLE
+                bind!!.runButtonLay.visibility = View.GONE
+                bind!!.pauseButtonLay.visibility = View.GONE
+                bind!!.countdown.text = "0"
+            }
+            TimerState.RUNNING -> {
+                bind!!.initButtonLay.visibility = View.GONE
+                bind!!.runButtonLay.visibility = View.VISIBLE
+                bind!!.pauseButtonLay.visibility = View.GONE
+            }
+            TimerState.PAUSED -> {
+                bind!!.initButtonLay.visibility = View.GONE
+                bind!!.runButtonLay.visibility = View.GONE
+                bind!!.pauseButtonLay.visibility = View.VISIBLE
+            }
+        }
+    }
+
+    private fun loadTimer(circuit: CircuitObject) {
+        sets = circuit.sets!!
+        currentSet = sets
+        timeRest = circuit.rest!!
+        timeWork = circuit.work!!
+
+        bind!!.name.text = circuit.name
+    }
+
+    private fun workout() {
+        runningState = RunningState.WORK
+        bind!!.state.text = "Workout"
+        bind!!.set.text = "Set Number " + currentSet.toString()
```

Comment:
line 151 and 152 show an IDE warning

Reply:
Yeah cuz it's temporary, all these functions (workout, rest, isdone) I just made them temporarily. Will use a similar format to organize these actions but the names of the functions will change and what they do will also change.